src/index.ts

```diff
@@ -686,12 +686,32 @@ function getPublicKeyFor<T extends MessageTypes>(
   return recoverPublicKey(msgHash, msgParams.sig);
 }
 
-function padWithZeroes(number: string, length: number): string {
-  let myString = `${number}`;
-  while (myString.length < length) {
-    myString = `0${myString}`;
+/**
+ * Pads the front of the given hex string with zeroes until it reaches the
+ * target length. If the input string is already longer than or equal to the
+ * target length, it is returned unmodified.
+ *
+ * If the input string is "0x"-prefixed or not a hex string, an error will be
+ * thrown.
+ *
+ * @param hexString The hexadecimal string to pad with zeroes.
+ * @param targetLength The target length of the hexadecimal string.
+ * @returns The input string front-padded with zeroes, or the original string
+ * if it was already greater than or equal to to the target length.
+ */
+export function padWithZeroes(hexString: string, targetLength: number): string {
```

Comment:
Gotta say, I'm not too keen on adding more stuff to this API :thinking: We've already exposed an annoying amount of what would otherwise be internal functions.

I think it would be fine to test this via `concatSig`, remove the tests, and keep this private.

Reply:
Either that or delete this altogether and use `String.prototype.padStart`

Reply:
TIL `padStart` is a thing! I'm definitely going to use that.